motionplan/constraint.go

```diff
@@ -330,12 +337,31 @@ func newCollisionConstraint(
 
 	// create constraint from reference collision graph
 	constraint := func(state *State) bool {
-		internal, err := state.Frame.Geometries(state.Configuration)
-		if err != nil && internal == nil {
-			return false
+		var internalGeoms []spatial.Geometry
+		if state.Configuration != nil {
+			internal, err := state.Frame.Geometries(state.Configuration)
+			if err != nil && internal == nil {
+				return false
+			}
+			internalGeoms = internal.Geometries()
+		} else {
```

Comment:
[nit] making this a switch statement would let you dedent and be more readable

Reply:
Dedented via a `else if`

Preference for me is to use a `switch` only if there's more than one `else if`

Reply:
Yeah I have the same preference but there are 3 cases here.  Fine if you keep it as an if statement but you can flatten it
```
if state.Configuration != nil
else if `state.Position != nil
else return false
```

Reply:
Already done in the bucket of work I'm about to push up